client/azure.go

```diff
@@ -123,83 +150,81 @@ func ResponseWasNotFound(resp autorest.Response) bool {
+		config.subscriptionID)
+	client.Authorizer = autorest.NewBearerAuthorizer(config.servicePrincipalToken)
 
 	return &client, nil
 }
 
-func newDNSZonesClient(config AzureConfig) (*dns.ZonesClient, error) {
-	spt, err := newServicePrincipalToken(config, azure.PublicCloud.ResourceManagerEndpoint)
-	if err != nil {
-		return nil, microerror.Maskf(err, "creating service principal token")
-	}
-
-	client := dns.NewZonesClient(config.SubscriptionID)
-	client.Authorizer = autorest.NewBearerAuthorizer(spt)
+func newDNSZonesClient(config *azureClientConfig) (*dns.ZonesClient, error) {
+	client := dns.NewZonesClientWithBaseURI(
+		config.resourceManagerEndpoint,
+		config.subscriptionID)
+	client.Authorizer = autorest.NewBearerAuthorizer(config.servicePrincipalToken)
 
 	return &client, nil
 }
 
-func newInterfacesClient(config AzureConfig) (*network.InterfacesClient, error) {
-	spt, err := newServicePrincipalToken(config, azure.PublicCloud.ResourceManagerEndpoint)
-	if err != nil {
-		return nil, microerror.Maskf(err, "creating service principal token")
-	}
-
-	client := network.NewInterfacesClient(config.SubscriptionID)
-	client.Authorizer = autorest.NewBearerAuthorizer(spt)
+func newInterfacesClient(config *azureClientConfig) (*network.InterfacesClient, error) {
+	client := network.NewInterfacesClientWithBaseURI(
+		config.resourceManagerEndpoint,
+		config.subscriptionID)
+	client.Authorizer = autorest.NewBearerAuthorizer(config.servicePrincipalToken)
 
 	return &client, nil
 }
 
-func newVnetPeeringClient(config AzureConfig) (*network.VirtualNetworkPeeringsClient, error) {
-	spt, err := newServicePrincipalToken(config, azure.PublicCloud.ResourceManagerEndpoint)
-	if err != nil {
-		return nil, microerror.Maskf(err, "creating service principal token")
-	}
-
-	client := network.NewVirtualNetworkPeeringsClient(config.SubscriptionID)
-	client.Authorizer = autorest.NewBearerAuthorizer(spt)
+func newVnetPeeringClient(config *azureClientConfig) (*network.VirtualNetworkPeeringsClient, error) {
+	client := network.NewVirtualNetworkPeeringsClientWithBaseURI(
+		config.resourceManagerEndpoint,
+		config.subscriptionID)
+	client.Authorizer = autorest.NewBearerAuthorizer(config.servicePrincipalToken)
 
 	return &client, nil
 }
 
-func newServicePrincipalToken(config AzureConfig, scope string) (*adal.ServicePrincipalToken, error) {
-	oauthConfig, err := adal.NewOAuthConfig(azure.PublicCloud.ActiveDirectoryEndpoint, config.TenantID)
+func newServicePrincipalToken(config AzureConfig, env *azure.Environment) (*adal.ServicePrincipalToken, error) {
+	oauthConfig, err := adal.NewOAuthConfig(env.ActiveDirectoryEndpoint, config.TenantID)
 	if err != nil {
 		return nil, microerror.Maskf(err, "creating OAuth config")
 	}
 
-	return adal.NewServicePrincipalToken(*oauthConfig, config.ClientID, config.ClientSecret, scope)
+	return adal.NewServicePrincipalToken(
+		*oauthConfig,
+		config.ClientID,
+		config.ClientSecret,
+		env.ServiceManagementEndpoint)
```

Comment:
Can we please not do this? It reads awfully IMHO.

Reply:
done.

Reply:
You only fixed this in one place but the other places are still using the "undesired" style. So I would not like to wrap the function calls and I would also like to keep the error handling explicit. So instead of this. 

```
return foo(
  a, 
  b, 
  c, 
  d)
```

Lets please do this, **everywhere**. 

```
err := foo(a, b, c, d)
if err != nil {
  return microerror.Mask(err)
}

return nil
```